### PR TITLE
Adds a comb.

### PR DESCRIPTION
**Pull request by etherware-novice**
## About The Pull Request
Adds a comb you can use to comb your hair and the like. Or others. You get it from maintenence or the clothesdrobe.

all sprite credits to hyperion

## Why It's Good For The Game
RP purposes, looking cool, someone asked me to make this

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

comb_chat
comb500

</details>

## Changelog
:cl:
add: Adds a comb for combing hair, found in maints and the clothesdrobe
/:cl:

### CODE REVIEW

**Comment by EvilDragonfiend** [comment]
Sprite is cursed

File: code/game/objects/items/cosmetics.dm [hunk 1]
```diff
@@ -289,6 +289,72 @@
...
+
+	if(patient == barber)
+		return attack_self(patient)
+
+	if(barber.a_intent == INTENT_HARM || !hair_check(M, user))
+		return ..()
+
+	patient.hair_style = applied_hair
+	patient.update_hair()
+	patient.visible_message("<span class='notice'>[user] carefully sweeps [patient]'s hair.</span>", "<span class='notice'>[user] carefully reaches forward and sweeps your hair!</span>")
```
Comment (Penwin0) [inline code/game/objects/items/cosmetics.dm:328]: This shouldn't be instant

File: code/game/objects/items/cosmetics.dm [hunk 2]
```diff
@@ -289,6 +289,72 @@
...
+
+/obj/item/comb/attack(mob/living/M, mob/living/user)
+	var/mob/living/carbon/human/patient = M
+	var/mob/living/carbon/human/barber = user
+
+	if(patient == barber)
+		return attack_self(patient)
+
+	if(barber.a_intent == INTENT_HARM || !hair_check(M, user))
+		return ..()
```
Comment (Penwin0) [inline code/game/objects/items/cosmetics.dm:319]: Does this mean you just slap them if they have no hair?

Comment (etherware-novice) [reply code/game/objects/items/cosmetics.dm:319]: wait

**Comment by etherware-novice** [comment]
> Sprite is cursed

i made it at like 1am

**Comment by LodedDiper69** [comment]
Seems a bit large for an item that is supposed to fit in your pocket.

**Comment by etherware-novice** [comment]
> Seems a bit large for an item that is supposed to fit in your pocket.

what about like the mod tablet sprite

**Comment by EvilDragonfiend** [comment]
> > Seems a bit large for an item that is supposed to fit in your pocket.
> 
> what about like the mod tablet sprite

check sprites of other barber-related items (razor, mirror, etc etc). It's too way big.

**Comment by DerpiestBruh** [comment]
la creatura

File: code/_globalvars/lists/maintenance_loot.dm [hunk 3]
```diff
@@ -114,5 +114,6 @@ GLOBAL_LIST_INIT(maintenance_loot, list(
 	/obj/item/throwing_star = 1,
 	/obj/item/circuitboard/machine/allaccess = 2,
 	/obj/item/pen/screwdriver = 8,
+	/obj/item/comb = 10,
```
Comment (Penwin0) [inline code/_globalvars/lists/maintenance_loot.dm:117]: Isn't this extremely common?

File: code/game/objects/items/cosmetics.dm [hunk 4]
```diff
@@ -289,6 +289,76 @@
...
+
+	if(patient == barber)
+		return attack_self(patient)
+	if(barber.a_intent == INTENT_HARM )
+		return ..()
+	if(!hair_check(M, user))
+		return
+
+	barber.visible_message("<span class='notice'>You attempt to sweep [patient]'s hair.</span>", "<span class='notice'>[barber] reaches towards [patient]'s hair with a comb.</span>")
+	if(!do_after(user, 3, target=patient))
```
Comment (Penwin0) [inline code/game/objects/items/cosmetics.dm]: This is pretty much instantaneous

File: code/modules/vending/clothesmate.dm [hunk 5]
```diff
@@ -7,7 +7,8 @@
 	product_slogans = "Dress for success!;Prepare to look swagalicious!;Look at all this swag!;Why leave style up to fate? Use the ClothesMate!"
 	vend_reply = "Thank you for using the ClothesMate!"
 	light_color = LIGHT_COLOR_GREEN
-	products = list(/obj/item/clothing/head/beanie = 3,
+	products = list(/obj/item/comb = 10,
```
Comment (Penwin0) [inline code/modules/vending/clothesmate.dm]: You have a bunch of clothing and then ten combs? you should just place a couple in bathrooms/barbershop instead

File: code/_globalvars/lists/maintenance_loot.dm (hunk 3 above)
Comment (etherware-novice) [reply code/_globalvars/lists/maintenance_loot.dm:117]: yes

Comment (PowerfulBacon) [reply code/_globalvars/lists/maintenance_loot.dm:117]: why would maints be filled with more combs than screwdrivers?

File: code/modules/vending/clothesmate.dm (hunk 5 above)
Comment (PowerfulBacon) [inline code/modules/vending/clothesmate.dm]: Remove the comb from this, its a clothes vendor

File: code/game/objects/items/cosmetics.dm [hunk 6]
```diff
@@ -289,6 +289,76 @@
...
+/obj/item/comb/attack_self(mob/user)
+	. = ..()
+	if(!hair_check(user, user))
+		return
+
+	var/mob/living/carbon/human/coolguy64 = user
+
+	coolguy64.hair_style = applied_hair
+	coolguy64.update_hair()
+	coolguy64.visible_message("<span class='notice'>[coolguy64] sweeps their hair back.</span>", "<span class='notice'>You sweep your hair back.</span>")
```
Comment (PowerfulBacon) [inline code/game/objects/items/cosmetics.dm:340]: Variable naming

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
Stopped reviewing here due to unaddressed reviews

File: code/_globalvars/lists/maintenance_loot.dm (hunk 3 above)
Comment (etherware-novice) [reply code/_globalvars/lists/maintenance_loot.dm:117]: hair specialists

File: code/game/objects/items/cosmetics.dm (hunk 6 above)
Comment (etherware-novice) [reply code/game/objects/items/cosmetics.dm:340]: how is this a varname issue its a cool guy cant i have little jokes in my code?!

**Pull request closed without merging**